Pylint alerts corrections as part of an intervention experiment 401

evidencebp opened the pull request:
Makes the interventions describe in [intervention issue](https://github.com/SublimeText/PackageDev/issues/401).
The experiment is described [here](https://github.com/evidencebp/pylint-intervention/).

Each intervention was done in a dedicated commit with a message explaining it.

The PR is still WIP.
I'd like to discuss some points in the issue.

FichteFoll reviewed (changes requested):
Thanks for taking a shot at this.

Unfortunately, the two large refactoring commits [82a16c6](https://github.com/SublimeText/PackageDev/pull/402/commits/82a16c67fcfa47f9be671b587afc046c0ee14f26) and [6b23492](https://github.com/SublimeText/PackageDev/pull/402/commits/6b2349272fde43fbd992a4c3a3f02727e96aff9d) introduced quite a few logic errors that were also not easy to spot during review and some of the function names choices are a bit unfortunate, which is likely because you're not familiar with the code and the problem domain (Sublime Text plugin). That is fine and all, but this review took me 45 minutes and that was honestly a bit more than I was anticipating.

I suggest to drop these two commits (or open a new PR without these two) and I will take a jab at refactoring the two functions myself. However, the `syntax_dev_legacy.py` file is, as the name indicates, in legacy territory and in maintenance mode, meaning I'm not really keen on spending time on it compared to addressing some of the already open issues. 

Regardless, I hope you can put my detailed comments to use in your research.

evidencebp commented:
> Thanks for taking a shot at this.
> 
> Unfortunately, the two large refactoring commits [82a16c6](https://github.com/SublimeText/PackageDev/pull/402/commits/82a16c67fcfa47f9be671b587afc046c0ee14f26) and [6b23492](https://github.com/SublimeText/PackageDev/pull/402/commits/6b2349272fde43fbd992a4c3a3f02727e96aff9d) introduced quite a few logic errors that were also not easy to spot during review and some of the function names choices are a bit unfortunate, which is likely because you're not familiar with the code and the problem domain (Sublime Text plugin). That is fine and all, but this review took me 45 minutes and that was honestly a bit more than I was anticipating.
> 
> I suggest to drop these two commits (or open a new PR without these two) and I will take a jab at refactoring the two functions myself. However, the `syntax_dev_legacy.py` file is, as the name indicates, in legacy territory and in maintenance mode, meaning I'm not really keen on spending time on it compared to addressing some of the already open issues.
> 
> Regardless, I hope you can put my detailed comments to use in your research.

I did as you suggested, @FichteFoll.
I reverted the two commits.
Sure, if the legacy is out of scope, no need to take care of it.
I'll be happy to get the reference to a PR doing the second fix.

Note that I also changed "pass" to "raise NotImplementedError" in loaders and dumpers.

Thank you for your help, @FichteFoll !

FichteFoll commented:
Thanks again. I have cherry-picked the non-problematic commits into another PR at #403 and merged it.
